### DISCUSSION TITLE
TypeScript configuration question

### DISCUSSION THREAD

**Discussion by burgessa23** [hrsh7th/nvim-cmp#680] [Q&A] [answered]
Hi,

With the new abilities in [`typescript-language-server`](https://github.com/typescript-language-server/typescript-language-server/pull/303) to provide snippets in completions, and [this](https://github.com/hrsh7th/nvim-cmp/issues/646) associated fix, it might be useful to provide exact configs for enabling this functionality.

Passing the preferences object into `init-options` seems to not have any effect for me.

```js
lsp.tsserver.setup {
  capabilities = capabilities,
  on_attach = on_attach,
  init_options = {
    preferences = {
      includeCompletionsWithSnippetText = true,
      includeCompletionsForImportStatements = true,
    },
  },
}
```

I thought this would enable similar placeholder text to what the Lua LSP provides.

Does anyone have any tips to get placeholder text working with typescript?

**Comment by hrsh7th** [comment]
I don't know... I think you should ask to typescript-language-server repo.

**Comment by gegoune** [comment, marked as answer]
Try:

```lua
lsp.tsserver.setup {
    settings = {
      completions = {
        completeFunctionCalls = true,
      },
    },
},
```

**Reply by burgessa23** [reply]
Thanks, this was the missing piece.
